### PR TITLE
Pod controller: "when Using pod group Should keep the running pod group with the queue name if workload is evicted" flakes

### LINKED ISSUE #2243: Pod controller: "when Using pod group Should keep the running pod group with the queue name if workload is evicted" flakes
/kind flake
**What happened**:
Integration Test failed on "Pod Controller Suite: [It] Pod controller when manageJobsWithoutQueueName is disabled when Using pod group Should keep the running pod group with the queue name if workload is evicted" 

```shell
{Timed out after 5.001s.
The function passed to Eventually failed at /home/prow/go/src/sigs.k8s.io/kueue/test/util/util.go:685 with:
Expected
    <[]v1.PodSchedulingGate | len:1, cap:1>: [
        {
            Name: "kueue.x-k8s.io/admission",
        },
    ]
not to contain element matching
    <v1.PodSchedulingGate>: {
        Name: "kueue.x-k8s.io/admission",
    } failed [FAILED] Timed out after 5.001s.
The function passed to Eventually failed at /home/prow/go/src/sigs.k8s.io/kueue/test/util/util.go:685 with:
Expected
    <[]v1.PodSchedulingGate | len:1, cap:1>: [
        {
            Name: "kueue.x-k8s.io/admission",
        },
    ]
not to contain element matching
    <v1.PodSchedulingGate>: {
        Name: "kueue.x-k8s.io/admission",
    }
In [It] at: /home/prow/go/src/sigs.k8s.io/kueue/test/integration/controller/jobs/pod/pod_controller_test.go:705 @ 05/21/24 05:50:59.975
}
```

**What you expected to happen**:
No error happend.

**How to reproduce it (as minimally and precisely as possible)**:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/2221/pull-kueue-test-integration-main/1792793854399746048

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Kueue version (use `git describe --tags --dirty --always`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by vladikkuzn**
#### What type of PR is this?
/kind failing-test

#### What this PR does / why we need it:
Fixes failing test caused by changes in isPodRunnableOrSucceeded of #2212
```
-	if p.DeletionTimestamp != nil && len(p.Spec.SchedulingGates) > 0 {
+	if p.DeletionTimestamp != nil && len(p.Spec.NodeName) == 0 {
```
It's solely tests issue, since node name was not set, the workload reconciliation was stuck in a reclaimable pods reset attempt

#### Which issue(s) this PR fixes:

Fixes #2243

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: test/integration/controller/jobs/pod/pod_controller_test.go [hunk 1]
```diff
@@ -697,6 +701,17 @@ var _ = ginkgo.Describe("Pod controller", ginkgo.Ordered, ginkgo.ContinueOnFailu
...
 
+					util.SetNodeName(ctx, k8sClient, "node1", replacementPod)
+
+					// Eventually pod2 fully removed
+					gomega.Eventually(func(g gomega.Gomega) bool {
+						err := k8sClient.Get(ctx, pod2LookupKey, createdPod)
+						if err != nil {
+							return apierrors.IsNotFound(err)
+						}
+						return createdPod.DeletionTimestamp.IsZero()
```
Comment (trasc) [inline test/integration/controller/jobs/pod/pod_controller_test.go:712]: we should wait for "not found" only, we have already establish that `createdPod.DeletionTimestamp` is not zero  a couple of lines above for this, making this wait useless.

File: test/integration/controller/jobs/pod/pod_controller_test.go [hunk 2]
```diff
@@ -653,6 +653,10 @@ var _ = ginkgo.Describe("Pod controller", ginkgo.Ordered, ginkgo.ContinueOnFailu
 					))
 				})
 
+				ginkgo.By("Set NodeName instead of scheduler", func() {
+					util.SetNodeName(ctx, k8sClient, "node1", pod2)
+				})
+
 				ginkgo.By("set the pods as running", func() {
 					util.SetPodsPhase(ctx, k8sClient, corev1.PodRunning, pod1, pod2)
 				})
```
Comment (trasc) [inline test/integration/controller/jobs/pod/pod_controller_test.go:662]:
```suggestion
				ginkgo.By("set the pods as running", func() {
					util.SetPodsPhase(ctx, k8sClient, corev1.PodRunning, pod1, pod2)
					util.SetNodeName(ctx, k8sClient, "node1", pod1, pod2)
				})
```

File: pkg/controller/jobs/pod/pod_controller.go
Comment (trasc) [inline pkg/controller/jobs/pod/pod_controller.go:1]: drop the changes in this file

**Review by trasc — COMMENTED** [review]
just some clean-up is needed, otherwise LGTM

**Review by trasc — COMMENTED** [review]
/lgtm

**Comment by alculquicondor** [comment]
In general, please explain (in the description) what was the sequence of events that lead to the flakiness and why this PR fixes the problem.

File: test/integration/controller/jobs/pod/pod_controller_test.go [hunk 3]
```diff
@@ -655,6 +655,7 @@ var _ = ginkgo.Describe("Pod controller", ginkgo.Ordered, ginkgo.ContinueOnFailu
 
 				ginkgo.By("set the pods as running", func() {
 					util.SetPodsPhase(ctx, k8sClient, corev1.PodRunning, pod1, pod2)
+					util.SetNodeName(ctx, k8sClient, "node1", pod1, pod2)
```
Comment (alculquicondor) [inline test/integration/controller/jobs/pod/pod_controller_test.go:658]:
```suggestion
					util.AssignPodToNode(ctx, k8sClient, "node1", pod1, pod2)
```

Also, just to match the sequence of changes in a production environment, first assign to the node and then set the phase.

Comment (alculquicondor) [reply test/integration/controller/jobs/pod/pod_controller_test.go]: yes, that works too

**Comment by alculquicondor** [comment]
/lgtm
/approve
/hold
in case you want to do the name change.

**Review by trasc — COMMENTED** [review]
/lgtm

**Pull request merged**